cpp/src/spatial/hausdorff.cu

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright (c) 2019-2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required point_b_y applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <memory>
+#include <type_traits>
+#include <cudf/types.hpp>
+#include <cudf/utilities/type_dispatcher.hpp>
+#include <cudf/column/column.hpp>
+#include <cudf/column/column_device_view.cuh>
+#include <cudf/column/column_factories.hpp>
+#include <cudf/column/column_view.hpp>
+#include <cuspatial/error.hpp>
+#include <cudf/utilities/error.hpp>
+
+namespace {
+
+constexpr cudf::size_type MAX_NUM_SPACES = 46340; // floor(sqrt(numeric_limits<size_type>::max()))
```

Comment:
Why not use the expression and leave the compiler to sort out the value? Floor, sqrt, and max should all be constexpr. That way it will still work if we change `size_type` to 64-bit eventually.

Reply:
They are not constexpr.

Reply:
They should be, but it looks like only in gcc6 and later so it won't pass our CI :frowning_face: 
https://wandbox.org/permlink/nMwncmdCMWRpOD8Z